components/merkledb/src/proto/proof.proto

```diff
@@ -0,0 +1,37 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+import "types.proto";
+import "google/protobuf/empty.proto";
+
+package exonum.proof;
+
+message MapProof {
+    repeated OptionalEntry entries = 1;
+    repeated MapProofEntry proof = 2;
+}
+
+message OptionalEntry {
+    bytes key = 1;
+    oneof maybe_value {
+        bytes value = 2;
+        google.protobuf.Empty no_value = 3;
+    }
+}
+
+message MapProofEntry {
+    bytes proof_path = 1;
+    exonum.crypto.Hash hash = 2;
+}
```

Comment:
It would be good to add some descriptions to messages and fields. A client code can use it to generate docs.